### DISCUSSION TITLE
images doesn't show

### DISCUSSION THREAD

**Discussion by dveretennikov1** [outline/outline#3281] [Self Hosting] [answered]
Hi, I'm trying to upload a profile photo. Got a message Profile picture updated, but images doesn't show
Images are created in minio and I could see it in bucket

I tried use minio with nginx and without, the same result
config to use without nginx
minio server /data/minio/ --console-address outlinedata.xxx.xxx:9001 --address outlinedata.xxx.xxx:9000 --certs-dir /etc/minio/certs/
API: https://outlinedata.xxx.xxx:9000
RootUser: minioadmin
RootPass: minioadmin
Region:   ap-southeast-1

Console: https://outlinedata.xxx.xxx:9001
RootUser: minioadmin
RootPass: minioadmin

AWS_ACCESS_KEY_ID=minioadmin
AWS_SECRET_ACCESS_KEY=minioadmin
AWS_REGION=ap-southeast-1
#AWS_S3_ACCELERATE_URL=
AWS_S3_UPLOAD_BUCKET_URL=https://outlinedata.xxx.xxx:9000
AWS_S3_UPLOAD_BUCKET_NAME=outbucket
AWS_S3_UPLOAD_MAX_SIZE=26214400
AWS_S3_FORCE_PATH_STYLE=true
AWS_S3_ACL=private

![Screenshot 2022-03-24 163434](https://user-images.githubusercontent.com/102303381/159930074-0545d09f-5e21-4609-9fdf-ef94a8631d8e.png)

**Comment by almereyda** [comment, marked as answer]
I've had a similar issue with using Minio, and had to set the `read` bucket policy to allow third-parties to view the contents of the bucket without logging in.

Additionally, I remember that there was a setting somewhere to have Outline decide to proxy the images uploaded to it, or to direct to the actual source of a file on the S3 service in use, which I had chosen. Thus the prior policy had to be set for unauthenticated clients, such as a user's web browser, which does not know the credentials provided to the Outline web service.

> *Edit:* Ah, no, that was Funkwhale.

    mc policy set download minio/outline

When you open the Network tab of your browser's developer tools while loading that page, does it reveal the kind of error that results in the image not being loaded?

The code in

https://github.com/outline/outline/blob/8aa25fd7d6c8f830ef2fdb8bec3399735db21441/.env.sample#L30-L46

suggests us to visit https://wiki.generaloutline.com/share/125de1cc-9ff6-424b-8415-0d58c809a40f for more details.

Even though the `S3_ACL` variable is not explained anywhere, which is another culprit for some other place, we also find here an instruction to

> Ensure that “Block public access” is set to “off” at the bucket level.

Maybe this helps already?

Else the page contains further details on how to configure CORS and how to create the proper IAM policy for a bucket's user on Amazon S3. Minio setups would have to take similar measures.

In summary, this is what I did:

1. Create `outline.acl.json`:

```json
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Effect": "Allow",
      "Action": [
        "s3:*"
      ],
      "Resource": [
        "arn:aws:s3:::outline/*"
      ]
    }
  ]
}
```

2. Create policy from file

    ```
    mc admin policy add minio outline outline.acl.json
    ```

3. Create bucket

    ```
    mc mb minio/outline
    ```

4. Create user

    ```
    mc admin user add minio outline
    ```

6. Enable policy for user

    ```
    mc admin policy set minio outline user=outline
    ```

5. Enable `download` policy on bucket

    ```
    mc policy set download minio/outline
    ```

---

These steps may vary from installation to installation, but could maybe give hints at what's going wrong in your place?

**Reply by MetaBliss** [reply]
Hi thanks for providing the 6 steps in command line @almereyda , can you show me how I can create policy from file, enable policy for user and enable download policy on bucket from the Minio Admin GUI Console? 

This is the picture I see when I log in. 

<img width="1512" alt="image" src="https://user-images.githubusercontent.com/84720384/223497765-9cd8bf6d-e83e-4921-be21-da4d7ff35c87.png">

I would love to get your advice as I cannot paste images into Outline, I get an error which I've referenced here: https://github.com/outline/outline/discussions/5004#discussion-4937453  

> I've had a similar issue with using Minio, and had to set the `read` bucket policy to allow third-parties to view the contents of the bucket without logging in.
> 
> Additionally, I remember that there was a setting somewhere to have Outline decide to proxy the images uploaded to it, or to direct to the actual source of a file on the S3 service in use, which I had chosen. Thus the prior policy had to be set for unauthenticated clients, such as a user's web browser, which does not know the credentials provided to the Outline web service.
> 
> > _Edit:_ Ah, no, that was Funkwhale.
> 
> ```
> mc policy set download minio/outline
> ```
> 
> When you open the Network tab of your browser's developer tools while loading that page, does it reveal the kind of error that results in the image not being loaded?
> 
> The code in
> 
> https://github.com/outline/outline/blob/8aa25fd7d6c8f830ef2fdb8bec3399735db21441/.env.sample#L30-L46
> 
> suggests us to visit https://wiki.generaloutline.com/share/125de1cc-9ff6-424b-8415-0d58c809a40f for more details.
> 
> Even though the `S3_ACL` variable is not explained anywhere, which is another culprit for some other place, we also find here an instruction to
> 
> > Ensure that “Block public access” is set to “off” at the bucket level.
> 
> Maybe this helps already?
> 
> Else the page contains further details on how to configure CORS and how to create the proper IAM policy for a bucket's user on Amazon S3. Minio setups would have to take similar measures.
> 
> In summary, this is what I did:
> 
>     1. Create `outline.acl.json`:
> 
> 
> ```json
> {
>   "Version": "2012-10-17",
>   "Statement": [
>     {
>       "Effect": "Allow",
>       "Action": [
>         "s3:*"
>       ],
>       "Resource": [
>         "arn:aws:s3:::outline/*"
>       ]
>     }
>   ]
> }
> ```
> 
>     2. Create policy from file
>        ```
>        mc admin policy add minio outline outline.acl.json
>        ```
> 
>     3. Create bucket
>        ```
>        mc mb minio/outline
>        ```
> 
>     4. Create user
>        ```
>        mc admin user add minio outline
>        ```
> 
>     5. Enable policy for user
>        ```
>        mc admin policy set minio outline user=outline
>        ```
> 
>     6. Enable `download` policy on bucket
>        ```
>        mc policy set download minio/outline
>        ```
> 
> 
> These steps may vary from installation to installation, but could maybe give hints at what's going wrong in your place?

**Comment by dveretennikov1** [comment]
it works
mc policy set download minio/outline

thanks!!
